### DISCUSSION TITLE
Is there an unpause_pipeline step or how can I automatically unpause a pipeline created with set_pipeline

### DISCUSSION THREAD

**Discussion by meezaan** [concourse/concourse#6324] [Help & Support] [answered]
Hi there,

I'm using Concourse 6.7.1.

I programatically create a whole bunch of pipelines depending on external triggers using the set_pipeline step.

This works great, except the pipelines are all paused by default. Is there any way to create them as unpaused or unpause them. The resource (which the step replaces, I believe) had the unpause option as shown on https://github.com/concourse/concourse-pipeline-resource.

How can I do this without using the resource and just using the step, if it is possible?

Thank you.

**Comment by evanchaoli** [comment, marked as answer]
`set_pipeline` step should set pipeline with un-paused state, see https://github.com/concourse/concourse/blob/v6.7.1/atc/exec/set_pipeline_step.go#L232.

I have used `set_pipeline` step in many of my pipelines, never needed to unpause them after set.

**Reply by meezaan** [reply]
Thank you @evanchaoli. I will check again.

I was testing by creating the and destroying the same pipeline again and again. It is possible that I had paused it at some point and it came back in that state.

I will confirm - as you pointed out, the code shows otherwise

**Reply by meezaan** [reply]
@evanchaoli You are right. But interestingly enough if I add a job via set pipeline, then it gets removed because the pipeline no longer sets it, but then it gets added again via set pipeline - in this instance it comes back up as paused - but that I can do with, I think.
